backend/persistent/datastores/csv.js

```diff
@@ -0,0 +1,98 @@
+const alasql = require('alasql');
+import fetch from 'node-fetch';
+import papa from 'papaparse';
+import {type} from 'ramda';
+
+import {parseSQL} from '../../parse';
+
+// define type error thrown by this connector
+export function CSVError(url, errors) {
+      this.name = 'CSVError';
+      this.message = 'Failed to parse CSV file ' + url;
+
+      if (Error.captureStackTrace) {
+          Error.captureStackTrace(this, CSVError);
+      } else {
+          this.stack = new Error(this.message).stack;
+      }
+
+      this.url = url;
+      this.errors = errors;
+
+      if (errors && errors[0] && errors[0].message) {
+          this.message = errors[0].message;
+      }
+}
+CSVError.prototype = Object.create(Error.prototype);
+CSVError.prototype.constructor = CSVError;
+
+// the data parsed from CSV files is stored here
+const connectionData = {};
```

Comment:
In #362 One of the comments was to start using JSDocs style for our code.  Should we be doing this on the backend as well?